modules/aws-terraform-backend/iam.tf

```
@@ -185,3 +185,121 @@ resource "aws_iam_role_policy_attachment" "extra_roles" {
   policy_arn = aws_iam_policy.limited.arn
 }
 
+
+resource "aws_iam_role" "readonly_terraform_state" {
+  count = var.create_readonly_role ? 1 : 0
+
+  name = var.readonly_tfstate_access_role_name
```

This is not a read-only tfstate access role!!!!
```suggestion
  name = var.readonly_tfstate_access_role_name
```